app.js

```diff
@@ -0,0 +1,66 @@
+let form = document.querySelector('form');
+
+form.addEventListener('submit', async function (event) {
+    event.preventDefault();
+    clearUI();
+    console.log(form.username.value);
+
+    try {
+        let githubResponse = await fetchGithubRepos(form.username.value);
+        if (githubResponse.ok) {
+            displayRepos(githubResponse.repos);
+        } else {
+            displayWarningMessage(githubResponse.message);
+        }
+    } catch (err) {
+        displayGenericError("Something went wrong. Refresh and try again!");
+    }
+})
+
+async function fetchGithubRepos(username) {
+    let resp = await fetch(`https://api.github.com/users/${username}/repos`);
+    let jsonResp = await resp.json();
+    if (resp.ok) {
+        return {
+            ok: true,
+            repos: jsonResp
+                .filter(repo => repo.private === false)
+                .map(repo => repo.name)
+        }
+    } else {
+        return {
+            ok: false,
+            message: jsonResp.message
+        }
+    }
+}
+
+function displayRepos(repoNames) {
+    let ul = document.createElement('ul');
+
+    repoNames.forEach(repoName => {
+        let li = document.createElement('li');
+        li.innerHTML = repoName;
+        ul.appendChild(li);
+    })
+    document.body.appendChild(ul);
+}
+
+function displayWarningMessage(repoMessage) {
+    let paragraph = document.createElement('p');
+    paragraph.innerText = repoMessage;
+    paragraph.classList.add('textWarning');
+    document.body.appendChild(paragraph);
+}
+
+function displayGenericError(errorMessage) {
+    let paragraph = document.createElement('p');
+    paragraph.innerText = errorMessage;
+    paragraph.classList.add('textError');
+    document.body.appendChild(paragraph);
+}
```

Comment:
A lot of duplicate code. Let's find a way to have a single function, and then a parameter that controls whether the text is `warning` or `error`.